src/wallet/wallet.cpp

```diff
@@ -2167,10 +2182,16 @@ bool CWallet::SelectCoins(const vector<COutput>& vAvailableCoins, const CAmount&
             ++it;
     }
 
+    size_t nLimitAncestors = std::min(GetArg("-limitancestorcount", DEFAULT_ANCESTOR_LIMIT), GetArg("-limitdescendantcount", DEFAULT_DESCENDANT_LIMIT));
+    bool frejectLongChains = GetBoolArg("-rejectlongwalletchains", DEFAULT_REJECT_TOOLONG_CHAINS);
+
     bool res = nTargetValue <= nValueFromPresetInputs ||
-        SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 1, 6, vCoins, setCoinsRet, nValueRet) ||
-        SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 1, 1, vCoins, setCoinsRet, nValueRet) ||
-        (bSpendZeroConfChange && SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 0, 1, vCoins, setCoinsRet, nValueRet));
+        SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 1, 6, 0, vCoins, setCoinsRet, nValueRet) ||
+        SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 1, 1, 0, vCoins, setCoinsRet, nValueRet) ||
+        (bSpendZeroConfChange && SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 0, 1, 2, vCoins, setCoinsRet, nValueRet)) ||
+        (bSpendZeroConfChange && SelectCoinsMinConf(nTargetValue - nValueFromPresetInputs, 0, 1, 4, vCoins, setCoinsRet, nValueRet)) ||
```

Comment:
@gmaxwell We're already not guaranteed to find an optimal spend, this is just best-efforts anyway, so in my opinion it's not worth trying to optimize further for users using unusual values of these policy limits, which we already discourage changes to by hiding away in help-debug.

At any rate I think having explicit small values here is overall a win, in case someone were using non-default arguments for nLimitAncestors -- this ensures that the coin selection is always first going to try coins that have no ancestors/descendants.